code/game/objects/structures/shower.dm

```diff
@@ -0,0 +1,238 @@
+#define SHOWER_FREEZING "freezing"
+#define SHOWER_NORMAL "normal"
+#define SHOWER_BOILING "boiling"
+
+/obj/machinery/shower
+	name = "shower"
+	desc = "The HS-451. Installed in the 2550s by the Nanotrasen Hygiene Division."
+	icon = 'icons/obj/watercloset.dmi'
+	icon_state = "shower"
+	density = FALSE
+	use_power = NO_POWER_USE
+	var/on = FALSE
+	var/current_temperature = SHOWER_NORMAL
+	var/datum/looping_sound/showering/soundloop
+
+/obj/machinery/shower/Initialize()
+	. = ..()
+	soundloop = new(list(src), FALSE)
+
+/obj/machinery/shower/Destroy()
+	QDEL_NULL(soundloop)
+	return ..()
+
+/obj/machinery/shower/interact(mob/M)
+	on = !on
+	update_icon()
+	handle_mist()
+	add_fingerprint(M)
+	if(on)
+		START_PROCESSING(SSmachines, src)
+		soundloop.start()
+		wash_turf()
+		for(var/atom/movable/G in loc)
+			SEND_SIGNAL(G, COMSIG_COMPONENT_CLEAN_ACT, CLEAN_WEAK)
+			if(isliving(G))
+				var/mob/living/L = G
+				wash_mob(L)
+			else if(isobj(G)) // Skip the light objects
+				wash_obj(G)
+	else
+		soundloop.stop()
+		if(isopenturf(loc))
+			var/turf/open/tile = loc
+			tile.MakeSlippery(TURF_WET_WATER, min_wet_time = 5 SECONDS, wet_time_to_add = 1 SECONDS)
+
+/obj/machinery/shower/attackby(obj/item/I, mob/user, params)
+	if(I.tool_behaviour == TOOL_ANALYZER)
+		to_chat(user, "<span class='notice'>The water temperature seems to be [current_temperature].</span>")
+	else
+		return ..()
+
+/obj/machinery/shower/wrench_act(mob/living/user, obj/item/I)
+	to_chat(user, "<span class='notice'>You begin to adjust the temperature valve with \the [I]...</span>")
+	if(I.use_tool(src, user, 50))
+		switch(current_temperature)
+			if(SHOWER_NORMAL)
+				current_temperature = SHOWER_FREEZING
+			if(SHOWER_FREEZING)
+				current_temperature = SHOWER_BOILING
+			if(SHOWER_BOILING)
+				current_temperature = SHOWER_NORMAL
+		user.visible_message("<span class='notice'>[user] adjusts the shower with \the [I].</span>", "<span class='notice'>You adjust the shower with \the [I] to [current_temperature] temperature.</span>")
+		log_game("[key_name(user)] has wrenched a shower to [current_temperature] at ([x],[y],[z])")
```

Comment:
move this to the individual's combat log instead. also switch the XYZ at the end to use either COORD or AREACOORD instead.